app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts

```diff
@@ -28,14 +28,19 @@ const STATUS_SUCCESS_LAST = 299
 const STATUS_NOT_FOUND = 404
 /** HTTP status indicating that the server encountered a fatal exception. */
 const STATUS_SERVER_ERROR = 500
-/**
- * HTTP status indicating that the request was successful, but the user is not authorized to access
- */
+/** HTTP status indicating that the request was successful, but the user is not authorized to access. */
 const STATUS_NOT_AUTHORIZED = 401
 
 /** The number of milliseconds in one day. */
 const ONE_DAY_MS = 86_400_000
 
+/** The internal port of the JSON WebSocket server in a Cloud VM. */
+const JSON_WS_SERVER_PORT = 30001
```

Comment:
Maybe it's better to put these vars into env?

Reply:
possibly? i don't imagine we'd ever want to change it though, *although* it's definitely better if we didn't have to pass it in at all and instead the cloud should somehow automatically handle everything... one possible solution is to have cloud pass the appropriate query params in its reverse proxy configuration, then frontend doesn't have to do anything. @PabloBuchu thoughts? is this possible? (can `proxy_pass` redirect to `localhost:1234/?ls=ws://localhost:30001`?)